hadoop-hdds/docs/content/design/storage-distribution.md

```diff
@@ -0,0 +1,201 @@
+| **Old OM** | **New SCM** | SCM receives old proto. In `ServerSideTranslatorPB`, it checks if the new block list is empty; if so, it decodes using the old structure. |
+| **New OM** | **Old SCM** | OM will be using `getScmInfo()` to fetch SCM metadata layout version. If the feature is not finalized, OM sends the old structure.        |
+| **New OM** | **New SCM** | Fully upgraded. OM sends size-aware transactions and SCM processes them accordingly.                                                      |
+| **Old OM** | **Old SCM** | Legacy setup. Both components use old proto structure.                                                                                    |
+
+Also in SCM, while upgrading for an existing Ozone cluster, all existing block deletion transactions prior to DATA_DISTRIBUTION finalized will be ignored to update DeletedBlocksTransactionSummary when it's removed from SCM DB. 
+DeletedBlocksTransactionSummary only counts the transaction after DATA_DISTRIBUTION is finalized.
+Also in DN side, newly added metadata will be persisted only if the feature is finalized.
+
+---
+## Approach 2: CLI-based (Not Proceeding)
+
+A CLI-based approach was evaluated to compute detailed usage and pending deletion breakdown by analyzing offline OM and SCM database checkpoints and querying DataNodes.
+While it offers precise, up-to-date results and independence from Recon, it introduces significant operational overhead.
+
+This approach requires generating and processing large metadata snapshots, which can take hours in large-scale clusters.
+Given its complexity, dependency on manual execution, and high resource consumption, we have chosen not to proceed with the CLI-based solution and instead focus on enhancing Recon for better usability and integration.
+
+## Metrics Exposure and Time-Series Tracking
+
+While Recon provides a point-in-time view of pending deletions and storage distribution, it is equally important to track these metrics over time to understand trends and validate reclamation progress.  
+To enable this, components should expose metrics that can be scraped by Prometheus and visualized in Grafana.
+
+### Component-wise Metrics
+
+- **Ozone Manager (OM)**
+  - Recon has already methods that are available to calculate the following information.
+    - open key used space
+    - committed key used space 
+    - containerPreAllocated space
+    - pending deletion 
+  - In every OM db syncing, we can update these metrics values.
+
+- **Storage Container Manager (SCM)**
+  - The DeletedBlockLogStateManager is enhanced to aggregate these block sizes in-memory, providing a DeletedBlocksTransactionSummary that includes total transaction count, total block count, total block size, and total replicated block size for pending deletions. 
+  - This summary is rebuilt from persisted transaction data upon SCM startup or leader election.
```

Comment:
Does this mean we have to iterate the entire block deletion table on startup or leader change? That is probably not going to work given the large deletion backlogs we have seen in live clusters.